hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +64,197 @@ private Sinks() {
         return new SinkImpl<>(sinkName, metaSupplier);
     }
 
+    /**
+     * Returns a sink that updates items it receives into a Hazelcast
+     * {@code IMap} with the specified name and merge function. The function
+     * will be wrapped inside an {@link EntryProcessor}. The value extracted
+     * from input item will be inserted to the map if there is no value associated
+     * with the key, otherwise {@code mergeFn} will be called with the associated value
+     * as oldValue and the value which we want insert as newValue. The return value
+     * of the {@code mergeFn} will be the new value of the key.
+     *
+     *  @param mapName name of the map
+     *  @param toKeyFn  function that extracts key from the input item.
+     *  @param toValueFn  function that extracts value from the input item.
+     *  @param mergeFn merge function contains the merge logic. The function
+     *                 will be called with {@code mergeFn.apply(V oldValue, V newValue)}
+     *  @param <E> type of input item
+     *  @param <K> type of the key
+     *  @param <V> type of the value
+     */
+    public static <E, K, V> Sink<E> mergingMap(
```

Comment:
The naming should be discussed. Maybe `mapWithMerging` and `mapWithUpdating`

Reply:
Having them all prefixed with `map` seems like a good idea so that they're more discoverable.